src/controllers/midi/hss1394controller.cpp

```diff
@@ -40,16 +40,17 @@ void DeviceChannelListener::Process(const hss1394::uint8 *pBuffer, hss1394::uint
                 if (i + 2 < uBufferSize) {
                     note = pBuffer[i+1];
                     velocity = pBuffer[i+2];
-                    emit incomingData(status, note, velocity, timestamp);
+                    emit receiveShortMessage(status, note, velocity, timestamp);
                 } else {
                     qWarning() << "Buffer underflow in DeviceChannelListener::Process()";
                 }
                 i += 3;
                 break;
             default:
                 // Handle platter messages and any others that are not 3 bytes
-                QByteArray outArray((char*)pBuffer,uBufferSize);
-                emit incomingData(outArray, timestamp);
+                QByteArray outArray = QByteArray::fromRawData(
+                        reinterpret_cast<const char*>(pBuffer), uBufferSize);
+                emit receiveSysex(outArray, timestamp);
```

Comment:
> You might have already seen the "...does not live long enough" message wink

The HSS1394 library is Windows and macOS only, so I cannot actually test it.

Reply:
We don't need to test it. This is just totally unsafe code that will hit someone sooner or later. It's like leaving a loaded gun at a public place.

Reply:
I remember we already discussed that, it's not as bad as expected:

> A subsequent attempt to modify the contents of the returned QByteArray or any copy made from it will cause it to create a deep copy of the data array before doing the modification

Maybe it actually works. When switching contexts Qt needs to make a copy of the QByteArray anyway. Using a const-ref in a direct connection would then be zero-copy.

Reply:
I already reverted this. I am not going to make potentially dangerous changes to code I cannot test. This is tangential to the the topic of this PR anyway.